File: src/main/org/nlogo/prim/etc/_bothends.java
package org.nlogo.prim.etc ;

import org.nlogo.agent.AgentSet;
import org.nlogo.agent.Link;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _bothends
	extends Reporter
{
	@Override
	public Syntax syntax()
	{
		return Syntax.reporterSyntax
			( Syntax.TYPE_AGENTSET , "---L" ) ;
	}
	@Override
	public Object report( final Context context )
	{
		return report_1( context ) ;
	}
	public AgentSet report_1( final Context context )
	{
		return ( (Link) context.agent ).bothEnds() ;
	}
}

File: src/main/org/nlogo/prim/etc/_bothends.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Link
import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _bothends extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.AgentsetType, "---L")
  override def report(context: Context) =
    report_1(context)
  def report_1(context: Context) =
    context.agent.asInstanceOf[Link].bothEnds
}
